### PR TITLE
Issue 6320 2

### PR DESCRIPTION
**Pull request by rnveach**
Draft Issue #6320

Start of PR to looking into cases that regression couldn't find. CI is expected to fail under all mutations are handled.

Coding report: http://rveach.no-ip.org/checkstyle/pitest/40/pitest-coding/com.puppycrawl.tools.checkstyle.checks.coding/index.html

### CODE REVIEW

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckPatterns.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+         BAND_ASSIGN, BOR_ASSIGN, BSR_ASSIGN, BXOR_ASSIGN, DIV_ASSIGN, \
+         MINUS_ASSIGN, MOD_ASSIGN, PLUS_ASSIGN, SL_ASSIGN, SR_ASSIGN, STAR_ASSIGN, \
+         LAMBDA, TEXT_BLOCK_LITERAL_BEGIN, LAND, LITERAL_INSTANCEOF, GT, LT, GE, \
+         LE, EQUAL, NOT_EQUAL, UNARY_MINUS, UNARY_PLUS, INC, DEC, LNOT, BNOT, \
+         POST_INC, POST_DEC
+
+
+*/
+
+//non-compiled with javac: Compilable with Java14
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckPatterns.java:14]: TODO: Looking for method to remove non-compilable file.

**Comment by rnveach** [comment]
TODO List: (Will be hidden away once done)


@@ -0,0 +1,59 @@
- [ ] +AbstractSuperCheck.java.html:<td class='covered'><pre><span  class='survived'>            &#38;&#38; !isSameNameMethod(literalSuperAst)) {</span></pre></td></tr>
- [ ] +AbstractSuperCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (literalSuperAst.getType() == TokenTypes.LITERAL_SUPER</span></pre></td></tr>
- [ ] +AbstractSuperCheck.java.html:<td class='covered'><pre><span  class='survived'>        return sibling == null || !getMethodName().equals(sibling.getText());</span></pre></td></tr>
- [ ] +EqualsAvoidNullCheck.java.html:<td class='covered'><pre><span  class='survived'>        return tokenType == TokenTypes.CLASS_DEF</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                            &#38;&#38; isSameVariables(storedVariable, variable)</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                        == ast.getParent()) {</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (ast.getParent().getType() == TokenTypes.SWITCH_RULE</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (astIterator.getType() == childType</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (storedVariable != null &#38;&#38; isSameVariables(storedVariable, ast)) {</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>            result = findLastChildWhichContainsSpecifiedToken(</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        return ast.getType() == TokenTypes.LITERAL_IF</span></pre></td></tr>
- [ ] +FinalLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        return loop1 != null &#38;&#38; loop1 == loop2;</span></pre></td></tr>
- [ ] +HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>                &#38;&#38; ast.getType() == TokenTypes.PARAMETER_DEF) {</span></pre></td></tr>
- [ ] +HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>                &#38;&#38; firstChild.getType() == TokenTypes.IDENT) {</span></pre></td></tr>
- [ ] +HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>                &#38;&#38; methodAST.getType() == TokenTypes.METHOD_DEF</span></pre></td></tr>
- [ ] +HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (method.getType() == TokenTypes.METHOD_DEF) {</span></pre></td></tr>
- [ ] +HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>            return instanceFields.contains(field)</span></pre></td></tr>
- [ ] +HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (ignoreSetter &#38;&#38; ast.getType() == TokenTypes.PARAMETER_DEF) {</span></pre></td></tr>
- [ ] +HiddenFieldCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (type == TokenTypes.CLASS_DEF</span></pre></td></tr>
- [ ] +IllegalInstantiationCheck.java.html:<td class='covered'><pre><span  class='survived'>        return pkgName != null</span></pre></td></tr>
- [ ] +IllegalTypeCheck.java.html:<td class='covered'><pre><span  class='survived'>            else if (child.getType() == TokenTypes.TYPE_ARGUMENTS) {</span></pre></td></tr>
- [ ] +IllegalTypeCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (s.indexOf(&#39;.&#39;) == -1) {</span></pre></td></tr>
- [ ] +IllegalTypeCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (toVisitAst == null) {</span></pre></td></tr>
- [ ] +IllegalTypeCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (modifiers.getFirstChild() != null) {</span></pre></td></tr>
- [ ] +MagicNumberCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (ScopeUtil.isInCodeBlock(ast)) {</span></pre></td></tr>
- [ ] +MagicNumberCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (list.length == 0) {</span></pre></td></tr>
- [ ] +OneStatementPerLineCheck.java.html:<td class='covered'><pre><span  class='survived'>                        &#38;&#38; currentStatement.getPreviousSibling().getType() == TokenTypes.RESOURCES;</span></pre></td></tr>
- [ ] +OneStatementPerLineCheck.java.html:<td class='covered'><pre><span  class='survived'>                currentStatement.getPreviousSibling() != null</span></pre></td></tr>
- [ ] +OneStatementPerLineCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (countOfSemiInLambda.isEmpty()) {</span></pre></td></tr>
- [ ] +OneStatementPerLineCheck.java.html:<td class='covered'><pre><span  class='survived'>            multiline = !TokenUtil.areOnSameLine(prevSibling, ast)</span></pre></td></tr>
- [ ] +OneStatementPerLineCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (!hasResourcesPrevSibling &#38;&#38; isMultilineStatement(currentStatement)) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>                        &#38;&#38; ast.getParent().getType() != TokenTypes.LITERAL_CATCH) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (isAnonymousClassDef(ast)) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (toVisit == null) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>               &#38;&#38; parent.getType() != TokenTypes.CTOR_DEF</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>            &#38;&#38; lastChild.getType() == TokenTypes.OBJBLOCK;</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (definitionToken.getType() == TokenTypes.STATIC_INIT) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (elistToken != null &#38;&#38; ident.getText().equals(ast.getText())) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (lookForMethod &#38;&#38; containsMethod(nameToFind)</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (variableDeclarationFrame.getType() == FrameType.CLASS_FRAME) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (fieldUsageFrame.getType() == FrameType.BLOCK_FRAME) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (sibling != null &#38;&#38; isAssignToken(parent.getType())) {</span></pre></td></tr>
- [ ] +RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>        return left.getType() == right.getType() &#38;&#38; left.getText().equals(right.getText());</span></pre></td></tr>
- [ ] +UnnecessaryParenthesesCheck.java.html:<td class='covered'><pre><span  class='survived'>            || parent.getType() != TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR) {</span></pre></td></tr>
- [ ] +UnnecessaryParenthesesCheck.java.html:<td class='covered'><pre><span  class='survived'>        else if (type != TokenTypes.ASSIGN</span></pre></td></tr>
- [ ] +UnnecessaryParenthesesCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (firstChild != null &#38;&#38; firstChild.getType() == TokenTypes.LPAREN) {</span></pre></td></tr>
- [ ] +UnnecessaryParenthesesCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (type != TokenTypes.ASSIGN</span></pre></td></tr>
- [ ] +UnnecessaryParenthesesCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (type == TokenTypes.LAMBDA &#38;&#38; isLambdaSingleParameterSurrounded(ast)) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>                &#38;&#38; currentAst.getType() != TokenTypes.RCURLY) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>                &#38;&#38; currentNode.getType() == TokenTypes.LITERAL_CATCH) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>                &#38;&#38; currentStatementAst.getType() != TokenTypes.RCURLY) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>                else if (isChild(currentNode, variable)) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (currentNode.getType() == TokenTypes.LITERAL_IF) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (exprWithVariableUsage == null) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (curNode.getType() == ast.getType() &#38;&#38; curNode.getText().equals(ast.getText())) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (currentNodeType == TokenTypes.SLIST) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (!firstUsageFound) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (!isVarInOperatorDeclaration &#38;&#38; operator.getType() == TokenTypes.LITERAL_IF) {</span></pre></td></tr>
- [ ] +VariableDeclarationUsageDistanceCheck.java.html:<td class='covered'><pre><span  class='survived'>        while (result</span></pre></td></tr>

**Comment by romani** [comment]
I will be more sense to move all this survival to ignore list as we have in pitest.sh and address each of them one by one

**Comment by rnveach** [comment]
> more sense to move all this survival to ignore list 

I tried to do that but before, but it resulted in the following instead because there was too many to ignore.
https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L2246-L2247
This is the reason I am doing things this way.

**Comment by rnveach** [comment]
Latest master brought in the following mutants surviving:
````
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                        &#38;&#38; TokenUtil.isTypeDeclaration(parentAst.getParent().getType())) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                        &#38;&#38; isInsideLocalAnonInnerClass(grandParent);</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                        || parentAst.getType() == TokenTypes.OBJBLOCK</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                    .getParent().getParent().getType() == TokenTypes.OBJBLOCK) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (!variableDesc.isInstVarOrClassVar()) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>                if (parentAst.getType() == TokenTypes.ENUM_CONSTANT_DEF</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (packageName.isEmpty()) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>            if (scopeAst == null) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        else if (type == TokenTypes.LITERAL_NEW</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        else if (type == TokenTypes.LITERAL_NEW</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        for (int i = 0; i &#60; length &#38;&#38; patternClass.charAt(i) == classToBeMatched.charAt(i); ++i) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (TokenUtil.isOfType(ast, SCOPES)) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (blockContainingLocalAnonInnerClass == null) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (blockContainingLocalAnonInnerClass == null) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (blockContainingLocalAnonInnerClass == null) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (blockContainingLocalAnonInnerClass == null</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (isNonLocalTypeDeclaration(typeDeclAst)) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (outerClassQualifiedName == null) {</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        if (parentAst.getType() == TokenTypes.OBJBLOCK</span></pre></td></tr>
+UnusedLocalVariableCheck.java.html:<td class='covered'><pre><span  class='survived'>        return typeDeclAst.getParent().getType() != TokenTypes.SLIST</span></pre></td></tr>
````

**Comment by Vyom-Yadav** [comment]
> Latest master brought in the following mutants surviving:

I haven't seen so many surviving mutations for a single check in my life, I will start working on it.

**Comment by rnveach** [comment]
Its a brand new mutator we have been trying to add. See this PR to reproduce the mutations locally.

**Comment by rnveach** [comment]
Closing this as it was moved to the new PR and will be finished piece by piece.

**Pull request closed without merging**